traffic_ops/testing/api/v4/serverservercapability_test.go

```diff
@@ -222,6 +223,43 @@ func GetTestServerServerCapabilities(t *testing.T) {
 	}
 }
 
+func UpdateTestServerServerCapabilities(t *testing.T) {
+	var header http.Header
+
+	// Get server capability name and edit it to a new name
+	resp, _, err := TOSession.GetServerCapabilitiesWithHdr(header)
+	if err != nil {
+		t.Fatalf("Expected no error, but got %v", err.Error())
+	}
+	origName := resp[0].Name
+	newSCName := "sc-test"
+	resp[0].Name = newSCName
+
+	// Update server capability with new name
+	updateResponse, _, err := TOSession.UpdateServerCapabilityByName(origName, &resp[0])
+	if err != nil {
+		t.Errorf("cannot PUT server capability: %v - %v", err, updateResponse)
+	}
+
+	//To check whether the primary key change trickled down to server table
+	ssc, _, err := TOSession.GetServerServerCapabilitiesWithHdr(nil, nil, &newSCName, nil)
+	if err != nil {
+		t.Fatalf("cannot GET server capabilities assigned to servers by server capability name %v: %v", newSCName, err)
+	}
+	for i, s := range ssc {
+		if *s.ServerCapability != *ssc[i].ServerCapability {
```

Comment:
This will always be same. Basically, you're just comparing the element with what you get when you get the element by its index. IMO, the test should be like this:
1.) You already have a list of servers with assigned server capabilities; Keep a track of which servers are assigned to the capability in question
2.) Update the name of this server capability
3.) Get the list of servers associated with this (updated)capability again
4.) Make sure the list in 1 matches the list in 3; If not, throw error.
 
Also, missing nil checks before dereferencing pointers.

Reply:
Updated test case to correctly check the trickle of server capability name change.